core/services/addon/AddonManager.php

```diff
@@ -0,0 +1,131 @@
+use Exception;
+use ReflectionException;
+
+/**
+ * Class AddonManager
+ *
+ * @author  Brent Christensen
+ * @package EventEspresso\core\services\addon
+ * @since   $VID:$
+ */
+class AddonManager
+{
+    /**
+     * @var AddonCollection
+     */
+    private $addons;
+
+    /**
+     * @var IncompatibleAddonHandler
+     */
+    private $incompatible_addon_handler;
+
+    /**
+     * @var Psr4Autoloader
+     */
+    private $psr4_loader;
+
+    /**
+     * @var RegisterV1Addon
+     */
+    private $register_v1_addon;
+
+    /**
+     * @var ThirdPartyPluginHandler
+     */
+    private $third_party_plugin_handler;
+
+
+    /**
+     * AddonManager constructor.
+     *
+     * @param AddonCollection          $addons
+     * @param Psr4Autoloader           $psr4_loader
+     * @param RegisterV1Addon          $register_v1_addon
+     * @param IncompatibleAddonHandler $incompatible_addon_handler
+     * @param ThirdPartyPluginHandler  $third_party_plugin_handler
+     */
+    public function __construct(
+        AddonCollection $addons,
+        Psr4Autoloader $psr4_loader,
+        RegisterV1Addon $register_v1_addon,
+        IncompatibleAddonHandler $incompatible_addon_handler,
+        ThirdPartyPluginHandler $third_party_plugin_handler
+    ) {
+        $this->addons                     = $addons;
+        $this->psr4_loader                = $psr4_loader;
+        $this->register_v1_addon          = $register_v1_addon;
+        $this->incompatible_addon_handler = $incompatible_addon_handler;
+        $this->third_party_plugin_handler = $third_party_plugin_handler;
+    }
+
+
+    /**
+     * @throws Exception
+     */
+    public function initialize()
+    {
+        // set autoloaders for all of the classes implementing the legacy EEI_Plugin_API
+        // which provide helpers for EE plugin authors to more easily register certain components with EE.
+        EEH_Autoloader::register_autoloaders_for_each_file_in_folder(EE_LIBRARIES . 'plugin_api');
+    }
+
+
+    /**
+     * @throws Exception
+     */
+    public function loadAddons()
+    {
+        try {
+            $this->incompatible_addon_handler->deactivateIncompatibleAddons();
+            // legacy add-on API
+            do_action('AHEE__EE_System__load_espresso_addons');
+            // new add-on API that uses versioning
+            do_action(
+                'AHEE__EventEspresso_core_services_addon_AddonManager__initialize__addons',
+                $this->addons,
+                espresso_version()
+            );
+            // addons are responsible for loading their AddonApiVersion into the AddonCollection
+            foreach ($this->addons as $addon) {
+                if ($addon instanceof AddonApiVersion) {
+                    $this->registerAddon($addon);
+                }
+            }
+            $this->third_party_plugin_handler->loadPlugins();
+            do_action('AHEE__EE_System__load_espresso_addons__complete');
```

Comment:
this is pretty much the same logic that was in EE_System for loading add-ons, but has been extracted into multiple smaller classes for SRP

Reply:
Nice 👍